src/shared/evntrace.rs

```diff
@@ -0,0 +1,961 @@
+    [u8; 16],
+    LogInstanceGuid LogInstanceGuid_mut: GUID,
+    s s_mut: TRACE_LOGFILE_HEADER_u2_s,
+}}
+STRUCT!{struct TRACE_LOGFILE_HEADER {
+    BufferSize: ULONG,
+    u1: TRACE_LOGFILE_HEADER_u1,
+    ProviderVersion: ULONG,
+    NumberOfProcessors: ULONG,
+    EndTime: LARGE_INTEGER,
+    TimerResolution: ULONG,
+    MaximumFileSize: ULONG,
+    LogFileMode: ULONG,
+    BuffersWritten: ULONG,
+    u2: TRACE_LOGFILE_HEADER_u2,
+    LoggerName: LPWSTR,
+    LogFileName: LPWSTR,
+    TimeZone: TIME_ZONE_INFORMATION,
+    BootTime: LARGE_INTEGER,
+    PrefFreq: LARGE_INTEGER,
+    StartTime: LARGE_INTEGER,
+    ReservedFlags: ULONG,
+    BuffersLost: ULONG,
+}}
+pub type PTRACE_LOGFILE_HEADER = *mut TRACE_LOGFILE_HEADER;
+STRUCT!{struct TRACE_LOGFILE_HEADER32 {
+    BufferSize: ULONG,
+    u1: TRACE_LOGFILE_HEADER_u1,
+    ProviderVersion: ULONG,
+    NumberOfProcessors: ULONG,
+    EndTime: LARGE_INTEGER,
+    TimerResolution: ULONG,
+    MaximumFileSize: ULONG,
+    LogFileMode: ULONG,
+    BuffersWritten: ULONG,
+    u2: TRACE_LOGFILE_HEADER_u2,
+    LoggerName: ULONG32,
+    LogFileName: ULONG32,
+    TimeZone: TIME_ZONE_INFORMATION,
+    BootTime: LARGE_INTEGER,
+    PrefFreq: LARGE_INTEGER,
+    StartTime: LARGE_INTEGER,
+    ReservedFlags: ULONG,
+    BuffersLost: ULONG,
+}}
+pub type PTRACE_LOGFILE_HEADER32 = *mut TRACE_LOGFILE_HEADER32;
+STRUCT!{struct TRACE_LOGFILE_HEADER64 {
+    BufferSize: ULONG,
+    u1: TRACE_LOGFILE_HEADER_u1,
+    ProviderVersion: ULONG,
+    NumberOfProcessors: ULONG,
+    EndTime: LARGE_INTEGER,
+    TimerResolution: ULONG,
+    MaximumFileSize: ULONG,
+    LogFileMode: ULONG,
+    BuffersWritten: ULONG,
+    u2: TRACE_LOGFILE_HEADER_u2,
+    LoggerName: ULONG64,
+    LogFileName: ULONG64,
+    TimeZone: TIME_ZONE_INFORMATION,
+    BootTime: LARGE_INTEGER,
+    PrefFreq: LARGE_INTEGER,
+    StartTime: LARGE_INTEGER,
+    ReservedFlags: ULONG,
+    BuffersLost: ULONG,
+}}
+pub type PTRACE_LOGFILE_HEADER64 = *mut TRACE_LOGFILE_HEADER64;
+STRUCT!{struct EVENT_INSTANCE_INFO {
+    RegHandle: HANDLE,
+    InstanceId: ULONG,
+}}
+pub type PEVENT_INSTANCE_INFO = *mut EVENT_INSTANCE_INFO;
+UNION2!{union EVENT_TRACE_PROPERTIES_u {
+    [u8; 4],
+    AgeLimit AgeLimit_mut: LONG,
+    FlushThreshold FlushThreshold_mut: LONG,
+}}
+STRUCT!{struct EVENT_TRACE_PROPERTIES {
+    Wnode: WNODE_HEADER,
+    BufferSize: ULONG,
+    MinimumBuffers: ULONG,
+    MaximumBuffers: ULONG,
+    MaximumFileSize: ULONG,
+    LogFileMode: ULONG,
+    FlushTimer: ULONG,
+    EnableFlags: ULONG,
+    u: EVENT_TRACE_PROPERTIES_u,
+    NumberOfBuffers: ULONG,
+    FreeBuffers: ULONG,
+    EventsLost: ULONG,
+    BuffersWritten: ULONG,
+    LogBuffersLost: ULONG,
+    RealTimeBuffersLost: ULONG,
+    LoggerThreadId: HANDLE,
+    LogFileNameOffset: ULONG,
+    LoggerNameOffset: ULONG,
+}}
+pub type PEVENT_TRACE_PROPERTIES = *mut EVENT_TRACE_PROPERTIES;
+// TODO: The single item structs are defined with the ":" bitfield syntax
```

Comment:
Oh it looks like I simply forgot to remove the comment after I found the macro to handle bitfields.